lib/krb5/build_auth.c

```diff
@@ -73,10 +75,111 @@ make_etypelist(krb5_context context,
 	krb5_abortx(context, "internal error in ASN.1 encoder");
     free_EtypeList(&etypes);
 
-    ret = _krb5_add_1auth_data(context,
-                               KRB5_AUTHDATA_GSS_API_ETYPE_NEGOTIATION, &e, 0,
-                               auth_data);
+    ade.ad_type = KRB5_AUTHDATA_GSS_API_ETYPE_NEGOTIATION;
+    ade.ad_data = e;
+
+    ret = add_AuthorizationData(auth_data, &ade);
+
     krb5_data_free(&e);
+
+    return ret;
+}
+
+static krb5_error_code
+store_32_le(const uint32_t in, krb5_data *out)
+{
```

Comment:
This can be replaced with `_gss_mg_encode_le_uint32()`

Reply:
That didn't work because libkrb doesn't link against that lib. Temporarily I copied the implementation, not sure how to go about it.

Reply:
Copying the implementation is fine.

Reply:
You can just inline (as in, copy and paste) the code. Obviously it shouldn't have a `_gss` prefix in libkrb5.

Reply:
Thanks!